src/js/binaryen.jsoo-extern-pre.js

```diff
@@ -0,0 +1,7 @@
+//Provides: binaryen
+var binaryen = {};
```

Comment:
What are the differences between this and the main binaryen.js build? This line seems to be added, and also `-s WASM=0` - anything else? (If just those two then maybe we can add a build flag with an ifdef for each of them, which would avoid duplication.)

Also, what does this line do? I'm not sure why it's needed in your build but not the default.

Reply:
`WASM=0` is actually specified in the `binaryen_js` target, but even binaryen.js only uses the wasm version now.

A few things are going on with this line:
1. If a `var` matching `EXPORT_NAME` doesn't exist, the value is attached to the global, which doesn't work for js_of_ocaml
2. js_of_ocaml uses the `//Provides: ` syntax to treeshake values from the program

Other changes include:
1. `ENVIRONMENT` to only `web` and `worker` because js_of_ocaml tries to get smart about node globals and it breaks the usage.
2. Compiling to ES5, which was changed previously.
3. `EXPORT_ES6=0` because we need plain ES5 javascript until js_of_ocaml finishes their ES6 support
4. `NO_FILESYSTEM=1` - since node is not supported, this ensures there is no filesystem related code in the file that might make js_of_ocaml think it is running in node.

I'm happy to use ifdef or any other solution you want, this was just the patch that I had to generate the file we need.

Reply:
I see, thanks. Ok, then I think a new build flag would be best. It sounds like these are specific to js_of_ocaml, so perhaps literally a new flag `JS_OF_OCAML`? Then we'd ifdef based on that on each of the bullet points you mention 1-2 and 1-4. That would avoid duplication and make it clear what the differences actually are (comments on each would be good as well).

Reply:
Thanks! Should be updated now (I also updated this branch to with `main`)